src/NuGet.Clients/NuGet.VisualStudio.Implementation/SolutionExplorer/Items/MenuController.cs

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Windows;
+
+using Microsoft.Internal.VisualStudio.PlatformUI;
+using Microsoft.VisualStudio;
+using Microsoft.VisualStudio.ProjectSystem.VS.Tree.Dependencies.AttachedCollections;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Shell.Interop;
+
+namespace NuGet.VisualStudio.SolutionExplorer
+{
+    internal sealed class MenuController : IContextMenuController
```

Comment:
Mostly asking for future work on my end.... We have other menu GUIDs defined in our VSCT (eg [IDM_VS_CTXT_PACKAGEREFERENCE_GROUP](https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Clients/NuGet.Tools/NuGetTools.vsct#L270)). Should we move those to this interface & pattern?

Reply:
I don't think so. This code only selects which menu should be chosen for these items. Your `.vsct` file may in future work with these menus to add commands to transitive items (eg. converting a transitive package reference to a direct package reference), but that work would be separate.